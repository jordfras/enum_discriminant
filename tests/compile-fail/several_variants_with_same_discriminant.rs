use enum_discriminant::discriminant;

#[discriminant(u8)]
enum SeveralVariantsWithSameDiscriminant {
    Variant1 = 1,
    Variant2 = 1,
}

// The same error should be produced with repr
#[repr(u8)]
enum SeveralVariantsWithSameDiscriminantRepr {
    Variant1 = 1,
    Variant2 = 1,
}

fn main() {}
