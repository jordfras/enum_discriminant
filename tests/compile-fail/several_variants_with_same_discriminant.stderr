error[E0081]: discriminant value `1` assigned more than once
 --> tests/compile-fail/several_variants_with_same_discriminant.rs:4:1
  |
4 | enum SeveralVariantsWithSameDiscriminant {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
5 |     Variant1 = 1,
  |                - `1` assigned here
6 |     Variant2 = 1,
  |                - `1` assigned here

error[E0081]: discriminant value `1` assigned more than once
  --> tests/compile-fail/several_variants_with_same_discriminant.rs:11:1
   |
11 | enum SeveralVariantsWithSameDiscriminantRepr {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
12 |     Variant1 = 1,
   |                - `1` assigned here
13 |     Variant2 = 1,
   |                - `1` assigned here
